### PR TITLE
Buffs the chilling effects of void heretic powers, fixes void storm visuals not updating

### PR DESCRIPTION
**Pull request by VioletN**
## About The Pull Request

This PR changes the way the chilling effects of void heretics' Mansus Grasp and Mark of Void work, making them stronger overall and more on par with the effects of the other heretic paths. Instead of simply lowering the target's body temperature by a small amount once, the aforementioned abilities now apply a status effect called "void chill," which lasts several seconds and gradually decreases the target's body temperature during this time period. It also applies a very slight slowdown modifier, just to slow the target down a bit until their body temperature gets low enough for _that_ to start slowing them down too.

The version of this effect applied by the Grasp of Void lowers body temperature by 21 K (14 * TEMPERATURE_DAMAGE_COEFFICIENT) every half second for 8 seconds. Against a target wearing civilian clothes in a standard indoor environment, this will usually result in the target getting the green "too cold" warning by the time they get off the ground from the Mansus Grasp, which will then persist for a few seconds longer, causing some lingering slowdown but no noticable burn damage.

The version of this effect applied by the activation of the Mark of Void lowers body temperature by 30 K (20 * TEMPERATURE_DAMAGE_COEFFICIENT) every half second for 10 seconds. Against a target wearing civilian clothes in a standard indoor environment, this will usually result in the target getting the blue "too cold" warning by the time they get off the ground from the Mansus Grasp. Although they will soon get back into the green, this lingers for many seconds, and causes some burn damage.

I've also removed the part where these abilities also slightly reduced the temperature of the turf the target was on, because it didn't really do anything noteworthy, and led to some weird unwanted interactions with things like air alarms and firelocks.

Obviously, both of these are much stronger against targets wearing thermally insulated clothing without a thermal regulator, such as a MODsuit or firesuit, or against targets that are weak to cold temperatures, such as lizardpeople and slimepeople, similar to changelings' cryo sting.

This is the icon for the void chill status effect:

image

In conjunction with this, this PR also makes some improvements to the void storm summoned by ascended void heretics. The void storm now inflicts less direct burn damage, but lowers its victims' body temperatures much more in addition to doing oxygen damage. A non-heretic who remains in the storm for too long will inevitably freeze to death unless they are immune to cold, and even then they are in danger of choking. From testing, it takes about half a minute for a human to drop into crit from full health in a room-temperature area inside the storm, and the slowdown caused by freezing makes it more difficult to escape or to engage the heretic directly.

To accompany this, I have also fixed the void storm visuals so that after the ascended heretic leaves an area, the storm overlay in that area will revert back to the light snow overlay it had before the heretic entered the area. This makes it much more clear whether or not the heretic is actually in the area, and by extension, which areas are actually hazardous to be in. It does take about a second or two for the visuals to update, which I'm not entirely happy with; ideally it would be faster than this, but it's still preferable to them not updating at all.

## Why It's Good For The Game

Currently, the chilling effects applied by these void heretic powers are so miniscule that they might as well not even exist; lowering a person's body temperature by 20 K _once_ isn't going to have any noticeable effect, even if they are wearing a MODsuit or are weak to cold. Mobs' body temperature simply adjusts to the air around them too fast for this to be a hazard. The void storm is especially disappointing - it's conceptually a really cool ability with great atmosphere, but in practice standing in the void storm just makes the game go "hm, kinda chilly in here, innit?"

I love the idea of void heretics being associated with the cold of space, so I wanted to bring this more to the forefront, especially since the other heretic paths get really cool powers with their Mansus Grasps and mark abilities - setting groups of people on fire, instakilling cyborgs, giving yourself an orbital knife shield... Being able to mute people is pretty helpful but not especially cool.

In practice, these changes would allow for some interesting setups and ingenuity from void heretics. Void chill is helpful enough on its own, but what if you were to orchestrate some sort of disaster that required people to suit up in thermally insulated clothing? Or, if your target is an engineer, you could use the spacewalking powers provided to you by Aristocrat's Way to ambush them on the solars. This status effect would also make the void Mansus Grasp very suitable for hit-and-run attacks, such as in a scenario where you are being chased by multiple people and don't have time to stand around whacking someone with your funny knife until they die; you could tap someone with the Mansus Grasp, Void Phase away, and the lingering slowdown would make it easier to get away from them, especially if they were wearing a MODsuit at the time.

It took a lot of testing and fiddling with the numbers to come up with the ideal values for this, but I think the ones I've got here work pretty well. Using cryo sting as a point of comparison, I would say that void chill is generally weaker than cryo sting because it lasts for a much shorter amount of time, but it does kick in almost instantaneously, and is of course still accompanied by the muting effect. I think this fits very well with the generally quick and violent gameplay style of heretics.

## Changelog

:cl:
balance: Instead of directly lowering body temperature by a small amount, the Grasp of Void and Mark of Void heretic abilities now inflict a status condition called void chill, which steadily lowers the victim's body temperature and also inflicts a mild slowdown.
balance: The void storm summoned by ascended void heretics now inflicts less direct burn damage, but lowers body temperature much more strongly.
fix: The void storm visual overlays now properly update when the heretic exits an area, making them a more accurate indicator of which area the heretic is in.
/:cl:

### CODE REVIEW

**Review by MrMelbert — CHANGES_REQUESTED** [review]

**Comment by private-tristan** [comment]
due to the storm not dealing direct burn damage and only changing temp and choking, won't people with temp adaptation from genetics be practically immune (other than needing oxy meds)?

**Comment by VioletN** [comment]
> due to the storm not dealing direct burn damage and only changing temp and choking, won't people with temp adaptation from genetics be practically immune (other than needing oxy meds)?

So, I've done some thinking (and testing) regarding this - it appears that with just temperature adaptation, you still pass out from oxygen loss within a reasonable amount of time, even with internals on. Injecting yourself with a salbutamol medipen does essentially negate the oxygen damage, which I'm not fully satisfied with, so I might add back in a little bit of direct burn damage. My main concern is striking a good balance here, so I don't necessarily want to make the damage too overwhelming, but it should still be pretty threatening.

Although, my other thoughts are that it's generally pretty obvious when someone is immune to cold, and heretics can always  choose to simply focus on killing such people directly using their spells or other means - as an ascended heretic, they still take half damage from brute and burn and can use all of their spells without a focus, so they definitely don't have to entirely rely on the storm to deal with enemies. I like the idea of people having to prepare to enter the storm similar to how you would prepare to enter any environmentally hazardous area, but they also probably shouldn't be able to _completely_ negate it.

File: code/datums/status_effects/debuffs/debuffs.dm [hunk 1]
```diff
@@ -1281,6 +1279,36 @@
 	desc = "You are a Ghoul!"
 	icon_state = ALERT_MIND_CONTROL
 
+//VOID CHILL
+/datum/status_effect/void_chill
+	id = "void_chill"
```
[the remark this answers is not in the document]
Comment (VioletN) [reply code/datums/status_effects/debuffs/debuffs.dm]: I've gone ahead and moved all the heretic-related status effects into four separate files - one for the mark status effects, one for the ghoul status effect, one for miscellaneous debuffs, and one for miscellaneous buffs. Hopefully I got all of them, you're right that these status effect files are very messy.

**Comment by VioletN** [comment]
I went ahead and added a little bit of the direct burn damage back to void storm, since upon testing it didn't seem too overpowered, and seems to play nicely with the temperature damage.

**Pull request merged**